saito-core/src/core/util/rate_limiter.rs

```diff
@@ -0,0 +1,58 @@
+use std::collections::HashMap;
+
+#[derive(Debug, Clone)]
+pub struct RateLimiter {
+    limits: HashMap<u64, TokenCounter>,
```

Comment:
Should we use PeerIndex as key or Public Key ? (If a peer disconnects and connects back, it will have a different index)